From e7873c4d7ee771b1f975986cf81ecf31d692580f Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@gmail.com>
Date: Wed, 14 Jan 2026 10:30:55 +0100
Subject: [PATCH] strace: use bundled kernel headers

Build against the bundled kernel headers instead of using the current
kernel ones. This ensures strace is using the kernel headers it is
written against, and not a random one that may contain breaking uapi
changes (which happen from time to time).

Fixes build against 6.18 final and recent LTS/stable kernels that got
minor uapi breakages (rename of a 6.18 introduced #define and a struct
field).

The PON kernel 6.18.52 user headers changed struct io_uring_buf_reg to
add "min_left" and turn "resv" into __u32[5], which breaks strace 6.17's
CHECK_TYPE_SIZE(arg.resv, sizeof(uint64_t) * 3) static assertion:

  io_uring.c: error: static assertion failed:
  "Unexpected size of arg.resv (sizeof(uint64_t) * 3 expected).
   --enabled-bundled=yes configure option may be used to work around that."

Signed-off-by: lunatickochiya <125438787@qq.com>

--- a/package/devel/strace/Makefile
+++ b/package/devel/strace/Makefile
@@ -10,7 +10,7 @@
 
 PKG_NAME:=strace
 PKG_VERSION:=6.17
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=https://strace.io/files/$(PKG_VERSION)
@@ -31,8 +31,6 @@
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/nls.mk
 
-HOST_CFLAGS += -I$(LINUX_DIR)/user_headers/include
-
 CONFIGURE_VARS+= \
 	LDFLAGS_FOR_BUILD="$(HOST_LDFLAGS)" \
 	CPPFLAGS_FOR_BUILD="$(HOST_CPPFLAGS)" \
@@ -72,6 +70,7 @@
 	--with-libdw=$(if $(CONFIG_STRACE_LIBDW),yes,no) \
 	--with-libunwind=$(if $(CONFIG_STRACE_LIBUNWIND),yes,no) \
 	--enable-mpers=no \
+	--enable-bundled=yes \
 	--without-libselinux
 
 MAKE_FLAGS := \
